src/System.Slices/src/System/SpanEnumerators.cs

```diff
@@ -88,13 +96,7 @@ public void Dispose()
 
             public bool MoveNext()
             {
-                int nextItemIndex = _position + 1;
-                if (nextItemIndex < _span.Length)
-                {
-                    _position = nextItemIndex;
-                    return true;
-                }
-                return false;
+                return ++_position < _span.Length;
```

Comment:
EnumeratorObject is not allowed to do this. 
By the IEnumerator contract, MoveNext must have no sideeffects when it returns false. User can call MoveNext as many times as he wants and it should not suddenly start returning true.

NOTE: this is not a requirement for the "foreach" struct enumerator above since that is supposed to be only called from foreach pattern.